### DISCUSSION TITLE
Two level list of atoms

### DISCUSSION THREAD

**Discussion by alessio-zamparelli** [pmndrs/jotai#2466] [Q&A]
Hi, here is my question:
I'm building an application where the user can categorize a list of tasks and can sort priority of both category and items.

As an example:
- cat1
  - item1
  - item2
- cat2
  - item3
  - item4

I'm using for the level 0 (category) split atoms for the insert/move option but I'm struggling to use the same approach on the inner list, actually as a list of item (they all share same struct {name: string, description: string, guests: string[]})

My hope is to create the same easy move/delete/insert option on the category level and on the item level.

Any idea ?

**Comment by himself65** [comment]
I would say this depends on how you design the data structure; there are too many ways to do so...

## 1. Make it an atom inside of list

```ts
const listAtom = atom([
	{
		catalog: 'animal',
		items: atom([
			{},
			{},
		]),
	}
])
```

This would make it difficult to persist the data because you mix code and data

## 2. use weak map to use spiltAtom inside of downside level

```ts
const listAtom = atom([
	{
		catalog: 'animal',
		items: [
			{},
			{},
		],
	}
])
const itemSplitAtomWeakMap = new WeakMap()

const Items = ({ item }) => {
	const itemSplitAtom = itemSplitAtomWeakMap.has(item) ? itemSplitAtomWeakMap.get(itemSplitAtomWeakMap) : itemSplitAtomWeakMap.set(item, splitAtom(item))
}
```

but I would say this code won't work as expected because of obj ref changes...

## 3. Make your logic because this is not jotai duty but your application logic
